From 7214ccbb760c45aa66a38208e1154769f3bd0e13 Mon Sep 17 00:00:00 2001
From: Richard Walker <rwalkerands@users.noreply.github.com>
Date: Fri, 20 Aug 2021 16:13:12 +1000
Subject: [PATCH] Support utf8mb4 on drush site-install #2352

Support utf8mb4 character set on drush site-install. See #2352
---
 src/Sql/SqlMysql.php | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Sql/SqlMysql.php b/src/Sql/SqlMysql.php
index 33f53464ab..550a19b9b4 100644
--- a/src/Sql/SqlMysql.php
+++ b/src/Sql/SqlMysql.php
@@ -96,7 +96,7 @@ public function createdbSql($dbname, $quoted = false)
             $dbname = '`' . $dbname . '`';
         }
         $sql[] = sprintf('DROP DATABASE IF EXISTS %s;', $dbname);
-        $sql[] = sprintf('CREATE DATABASE %s /*!40100 DEFAULT CHARACTER SET utf8 */;', $dbname);
+        $sql[] = sprintf('CREATE DATABASE %s /*!40100 DEFAULT CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci */;', $dbname);
         $db_superuser = $this->getOption('db-su');
         if (isset($db_superuser)) {
             // - For a localhost database, create a localhost user.  This is important for security.
